nixos/modules/config/networking.nix

```diff
@@ -223,7 +223,9 @@ in
             '';
 
       } // optionalAttrs config.services.resolved.enable {
-        "resolv.conf".source = "/run/systemd/resolve/resolv.conf";
+        # symlink the static version of resolv.conf as recommended by upstream:
+        # https://www.freedesktop.org/software/systemd/man/systemd-resolved.html#/etc/resolv.conf
+        "resolv.conf".source = "${pkgs.systemd}/lib/systemd/resolv.conf";
```

Comment:
If I didn't miss anything (still a NixOS newbie!):
I don't think my commit changes the current situation and therefore should not break any setups: On NixOS, `resolved` can't be used currently with programs like `openresolv` that will try to modify `/etc/resolv.conf` themselves.

More details (as I think to understand them): Currently, `resolved.enable` also means that `/etc/resolv.conf` will be managed by `resolved`. This means that `resolved` can't be used in conjunction with other programs that dynamically try to add DNS servers to `/etc/resolv.conf` or try to manage that file themselves. Aside from `openresolv`, that's also the case for stuff like NetworkManager.

`resolved` actually could support `/etc/resov.conf` being managed by something else by *not* getting a symlink from `/etc/resolv.conf` to anything from `systemd-resolved`. But I don't think there's any option for this in current NixOS.

Programs can also let `resolved` manage `/etc/resolv.conf` and use `resolved`'s dbus API to dynamically add DNS servers.  Some programs support this, e.g. I created #27142 to make that possible for `NetworkManager` on NixOS, and therefore make it possible to run `NetworkManager` and `resolved` in conjunction on NixOS. They work regardless of whether the static or the dynamic symlink is used.